matlab_simulation/code v1.0/ME597-8-NLP/nonlinearrhc.m

```diff
@@ -75,59 +82,52 @@
 x0(5:N:end) = 0;
 
 % Receding horizon goal
-xd = xdT(1:T,:)
-
-
-ii=1;                               % figure name: Creating animation in one figure
-filename = 'NLP_Animation.gif';     % saving animation file in .gif file
+xd = xdT(1:T,:);
+ii=1;                                % figure name
+filename = 'NLP_Animation.gif';      % saving animation file in .gif file
 
 % Repeat optimization at each timestep
-or i=1:TTot-T
+for i=1:TTot-T
 
     % Solve nonlinear program
-    options = optimset('maxfunevals',50000);
-    tic;
+    options = optimset('maxfunevals',50000,'display','off');
     [X,FVAL,EXITFLAG,OUTPUT,LAMBDA] = fmincon(@(x) cost(x),x0,A,B,Aeq,Beq,LB,UB,@(x) constraints(x), options);
-    toc;
+   
     % Move and Update problem for next timestep
     x0(1:N*(T-1)) = X(N+1:N*T,1);
-    x0(N*(T-1)+1:N*T) = X(N*(T-1)+1:N*T); % should be propagating dynamics here.
+    x0(N*(T-1)+1:N*T) = X(N*(T-1)+1:N*T);   % should be propagating dynamics here.
     xd = xdT(i+1:i+T,:);
     Beq = X(N+1:N+n)';
 
     % Rename results
-    x = X(1:N:end);
-    y = X(2:N:end);
-    th = X(3:N:end);
-    v = X(4:N:end);
-    w = X(5:N:end);
+    x = X(1:N:end);     % position on x-axis
+    y = X(2:N:end);     % position on y-axis
+    th = X(3:N:end);    % orientation
+    v = X(4:N:end);     % translational velocity
+    w = X(5:N:end);     % angular velocity
 
     % Plot results
-%     figure(2);clf; hold all;
-%     plot(1:T,x)
-%     plot(1:T,y)
-%     plot(1:T,th)
-%     plot(1:T,v)
-%     plot(1:T,w)
-
     figure(ii); clf; hold on;
     plot(x,y,'bx-');
+    % draw vehicle as a two-wheeled car
+    drawcar(x(end),y(end),th(end),.15,ii)   
     if (~endonly)
         plot(xdT(1:end-1,1), xdT(1:end-1,2), 'ro--')
     else
         plot(pF(1),pF(2),'ro')
     end
+    % draw obstacles
     if (withobs)
         for j=1:numObsts
-            plot(obs(j,1), obs(j,2),'bx');
+            plot(obs(j,1), obs(j,2),'kx');
             circle(ii, obs(j,:), radius(j));
         end
         axis equal
     end
-    %drawnow();
     F(i) = getframe;
     
-    % saving final animation file as a .gif file
+    
+    % saving animation file in .gif file
```

Comment:
Animating using a .gif is cool, but can you make it a function in utilities that you call?